common/djangoapps/course_modes/views.py

```diff
@@ -112,8 +112,7 @@ def get(self, request, course_id, error=None):
         has_enrolled_professional = (CourseMode.is_professional_slug(enrollment_mode) and is_active)
         if CourseMode.has_professional_mode(modes) and not has_enrolled_professional:
             purchase_workflow = request.GET.get("purchase_workflow", "single")
-            verify_url = IDVerificationService.get_verify_location('verify_student_start_flow', course_id=course_key)
-            redirect_url = "{url}?purchase_workflow={workflow}".format(url=verify_url, workflow=purchase_workflow)
+            redirect_url = IDVerificationService.get_verify_location(course_id=course_key)
```

Comment:
In our case at EDUlib, we do have ecommerce configured. We even had to adapt the code for the CyberSource payment processor a few years ago in order to work with Netbanx / Paysafe because our bank was dealing with them instead of CyberSource. Our payment page is hosted at Paysafe after a verified certificate is bought by the student.

Your conditional statement checks if it is a professional mode. I hope you have also taken into account the "standard" verified mode certificates. I have always been under the impression that professional mode was a different mode. I could be mistaken though...

Reply:
@sambapete Got it. Thank you for your detailed explanation.
Yes, you are correct that the changes I made here are Professional modes. I take that you don't have Professional modes in EDUlib? For verified mode, our flow has already been redirecting the user to ecommerce service. The ID Verification link will then be shown at the order receipt page. This logic in my opinion is essentially obsolete and should be retired if possible. That might come later.